src/librustdoc/core.rs

```diff
@@ -52,7 +52,7 @@ crate struct DocContext<'tcx> {
     /// Later on moved into `cache`
     crate renderinfo: RefCell<RenderInfo>,
     /// Later on moved through `clean::Crate` into `cache`
-    crate external_traits: Rc<RefCell<FxHashMap<DefId, clean::Trait>>>,
+    crate external_traits: Rc<RefCell<FxHashMap<DefId, (clean::Trait, bool)>>>,
```

Comment:
I feel pretty strongly that a struct should be extracted here, for two reasons:

1. It's unclear what the `bool` represents — it could mean anything
2. It's very possible that we'll add more fields in the future, and we don't want to tack them on to the tuple

(This also applies for the other place `external_traits` is defined.)

Reply:
I hesitated for a while and thought it might be best to keep it like this. If I add a comment, is it fine for you or do you still prefer that I create a new type wrapping `clean::Trait` and the `bool`?

Reply:
> thought it might be best to keep it like this

Can you explain your thinking a bit?

> If I add a comment, is it fine for you or do you still prefer that I create a new type wrapping `clean::Trait` and the `bool`?

I still prefer a new type :)

Many people likely won't read the comment. Besides, what's wrong with creating a new type? If for some reason it's a problem we can always change it to a tuple. Having a field name makes it much clearer what the `bool` is for, makes it much harder to introduce a bug by misunderstanding the meaning, and means people will not have to search to find docs in order to interact with it.

Reply:
Well... It's yet another type and we're trying to reduce that number. But, well, I don't mind making a new type so let's go for it!

Reply:
If it helps, my motivation for removing types is not so much "more types are bad", but that we do a lot of shuffling around of the same data between different structures. Since this isn't copying the data, just refactoring out a type, I think it's a good change.